File: retrofit-example/package/domain/Car.php
<?php
/**
 * Created by PhpStorm.
 * User: viniciusthiengo
 * Date: 10/11/15
 * Time: 10:28 PM
 */

class Car extends Obj {
    public $brand;
    public $engine;


    public function __construct( $name, $pathImg, Brand $brand, Engine $engine ){
        parent::__construct($name, $pathImg);
        $this->brand = $brand;
        $this->engine = $engine;
    }
}

File: app/src/main/java/br/com/thiengo/retrofitexample/domain/Car.java
package br.com.thiengo.retrofitexample.domain;

import br.com.thiengo.retrofitexample.util.Obj;

/**
 * Created by viniciusthiengo on 10/11/15.
 */
public class Car extends Obj {
    private Brand brand;
    private Engine engine;


    public Car() {}

    public Car(String name, String pathImg, Brand brand, Engine engine) {
        super(name, pathImg);
        this.brand = brand;
        this.engine = engine;
    }


    public Brand getBrand() {
        return brand;
    }

    public void setBrand(Brand brand) {
        this.brand = brand;
    }

    public Engine getEngine() {
        return engine;
    }

    public void setEngine(Engine engine) {
        this.engine = engine;
    }
}
